woocommerce/woocommerce-ios
# REST API: Updated UI and logic for the email login screen

Part of #8918 


## Description

There's an edge case for users whose WPCom accounts are considered suspicious and must log in with their usernames instead of emails. We decided that the simplest solution for this case is to follow Calypso to display the email label and placeholder to include "username" and show an error alert when username login is required.

This PR also refactors `JetpackSetupCoordinator` to move the WPCom account check and authentication request logic to `WPComEmailLoginViewModel` to make the coordinator cleaner. The coordinator still needs to call these requests from some flows other than the email/username login, so the `checkWordPressComAccount` and `requestAuthenticationLink` methods are kept internal instead of private.


## Testing instructions

Prerequisite: Make sure that you have access to a site with WooCommerce and no Jetpack.
- Log out of the app or skip onboarding if needed.
- Tap "Enter your site address" on the prologue screen and proceed with the address of your test site.
- Log in with the credentials of an admin.
- When the login completes, tap the Jetback benefit banner at the bottom of the Dashboard screen.
- Tap Log In on the benefit modal, notice that the text field now has the label and placholder to include both email and username.
- Enter the email address of a WPCom account that is required to log in with the username. If you don't have access to one, try [p1677851120396959/1677500033.040359-slack-C046HDZL87J]. Notice that an error alert is displayed asking for username.
- Enter the email address of a non-suspicious account. You should be able to log in successfully and see `✅ Ready for Jetpack setup` in Xcode console.

## Screenshots

<img src="https://user-images.githubusercontent.com/5533851/223014106-7b2a58f8-4802-4145-b319-d6dca83423f1.png" width=320 />

<img src="https://user-images.githubusercontent.com/5533851/223014111-2ad33963-876b-495d-b4b4-1206a2bbc855.png" width=320 />
